forc-test/src/lib.rs

```diff
@@ -115,24 +117,117 @@ pub struct TestPrintOpts {
 
 /// The storage and the contract id (if a contract is being tested) for a test.
 #[derive(Debug)]
-struct TestSetup {
+enum TestSetup {
+    ContractSetup(ContractTestSetup),
+    NonContractSetup(vm::storage::MemoryStorage),
+}
```

Comment:
I wonder if we should remove this distinction?

E.g. a `script` type program might have contract dependencies and need the contract setup anyway? Maybe this should look something like what `ContractTestSetup` currently looks like, but with `root_contract_id` as an `Option` (i.e. `Some` if the root is a contract, `None` otherwise)? I.e.

```rust
struct TestSetup {
    storage: vm::storage::MemoryStorage,
    contract_dependency_ids: Vec<tx::ContractId>,
    root_contract_id: Option<tx::ContractId>,
}
```

For libraries and predicates that cannot have contract dependencies, perhaps the `contract_dependency_ids` `Vec` can just remain empty?